sdk/samples/iot/aziot_realtek_amebaD/aziot_realtek_amebaD.ino

```diff
@@ -0,0 +1,385 @@
+  };
+
+  *out_signed_signature = az_span_create(az_span_ptr(signed_signature), 32);
+}
+
+static void base64_encode_bytes(
+    az_span decoded_bytes,
+    az_span base64_encoded_bytes,
+    az_span* out_base64_encoded_bytes)
+{
+  size_t len;
+  if( mbedtls_base64_encode( az_span_ptr(base64_encoded_bytes), (size_t)az_span_size(base64_encoded_bytes), 
+    &len, az_span_ptr(decoded_bytes), (size_t)az_span_size(decoded_bytes)) != 0)
+  {
+    Serial.println("[ERROR] mbedtls_base64_encode fail");
+  }
+  
+  *out_base64_encoded_bytes = az_span_create(az_span_ptr(base64_encoded_bytes), (int32_t)len);
+}
+
+static void decode_base64_bytes(
+    az_span base64_encoded_bytes,
+    az_span decoded_bytes,
+    az_span* out_decoded_bytes)
+{
+  
+  memset(az_span_ptr(decoded_bytes), 0, (size_t)az_span_size(decoded_bytes));
+
+  size_t len;
+  if( mbedtls_base64_decode( az_span_ptr(decoded_bytes), (size_t)az_span_size(decoded_bytes), 
+      &len, az_span_ptr(base64_encoded_bytes), (size_t)az_span_size(base64_encoded_bytes)) != 0)
+  {
+    Serial.println("[ERROR] mbedtls_base64_decode fail");
+  }
+  
+  *out_decoded_bytes = az_span_create(az_span_ptr(decoded_bytes), (int32_t)len);
+}
+
+static void iot_sample_generate_sas_base64_encoded_signed_signature(
+    az_span sas_base64_encoded_key,
+    az_span sas_signature,
+    az_span sas_base64_encoded_signed_signature,
+    az_span* out_sas_base64_encoded_signed_signature)
+{
+  //IOT_SAMPLE_PRECONDITION_NOT_NULL(out_sas_base64_encoded_signed_signature);
+
+  // Decode the sas base64 encoded key to use for HMAC signing.
+  char sas_decoded_key_buffer[32];
+  az_span sas_decoded_key = AZ_SPAN_FROM_BUFFER(sas_decoded_key_buffer);
+  decode_base64_bytes(sas_base64_encoded_key, sas_decoded_key, &sas_decoded_key);
+
+  // HMAC-SHA256 sign the signature with the decoded key.
+  char sas_hmac256_signed_signature_buffer[32];
+  az_span sas_hmac256_signed_signature = AZ_SPAN_FROM_BUFFER(sas_hmac256_signed_signature_buffer);
+  hmac_sha256_sign_signature(sas_decoded_key, sas_signature, sas_hmac256_signed_signature, &sas_hmac256_signed_signature);
+
+  // Base64 encode the result of the HMAC signing.
+  base64_encode_bytes(
+    sas_hmac256_signed_signature,
+    sas_base64_encoded_signed_signature,
+    out_sas_base64_encoded_signed_signature);
+}
+
+static void generate_sas_key(void)
+{
+
+  az_result rc;
+  // Create the POSIX expiration time from input minutes.
+  uint64_t sas_duration = iot_sample_get_epoch_expiration_time_from_minutes(SAS_TOKEN_EXPIRY_IN_MINUTES);
+
+  char b64enc_hmacsha256_signature[64];
+
+  // Get the signature that will later be signed with the decoded key.
+  az_span sas_signature = AZ_SPAN_FROM_BUFFER(signature);
+  rc = az_iot_hub_client_sas_get_signature(
+    &hub_client, sas_duration, sas_signature, &sas_signature);
+  if (az_result_failed(rc))
+  {
+    Serial.print("Could not get the signature for SAS key: az_result return code ");
+    Serial.println(rc);
```

Comment:
```suggestion
    Serial.println(rc);
    return;
```

Reply:
Will leave as is to keep compatible with Realtek repo, where this code comes from.